File: src/taeeun/week 23, 24/마법의 엘리베이터.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int storey) {
    int answer = 0;
    while (storey) {
        int remain = storey%10;
        int next=(storey/10)%10;

        if (remain>5) {
            answer+=(10-remain);
            storey+=10;
        }
        else if (remain<5) {
            answer+=remain;
        }
        else if (remain==5) {
            answer+=remain;
            storey+=next>=5?10:0;
        }
        storey/=10;
    }

    return answer;
}

File: src/geonhyeong/week23~24/마법의 엘리베이터.swift
// 마법의 엘리베이터
// Question Link: https://school.programmers.co.kr/learn/courses/30/lessons/148653

import Foundation

func solution(_ storey:Int) -> Int {
	var cnt = 0
	var storey = storey
	
	while storey > 0 {
		let remain = storey % 10
		
		if remain < 5 {				// 0 ~ 4
			cnt += remain
		} else if remain > 5 {		// 6 ~ 9
			cnt += (10 - remain)
			storey += 10
		} else {					// 5
			if (storey / 10) % 10 > 4 {
				storey += 10
			}
			cnt += remain
		}
		storey /= 10
	}
	
	return cnt
}
